lib/components/DownloadsScreen/downloaded_items_list.dart

```diff
@@ -1,104 +1,168 @@
-                    ),
-                  IconButton(
-                      icon: const Icon(Icons.delete),
-                      onPressed: () => askBeforeDeleteDownloadFromDevice(
-                              context, album, refresh: () {
-                            setState(() {});
-                          })),
-                ],
-              ),
-              subtitle: ItemFileSize(
-                stub: album,
-              ),
-              tilePadding: const EdgeInsets.symmetric(horizontal: 4.0),
-              children: [
-                DownloadedChildrenList(
-                  parent: album,
-                )
-              ],
-            );
-          },
-          childCount: items.length,
+    if (!_supportedTypes.contains(widget.type)) {
+      return const SliverToBoxAdapter(
+        child: Padding(
+          padding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
+          child: Text('Unsupported type'),
         ),
-      ),
-    );
+      );
+    }
+
+    return ref
+        .watch(_downloadsService.downloadedItemsProvider(widget.type))
+        .when(
+          data: (items) => items.isNotEmpty
+              ? SliverList(
+                  delegate: SliverChildBuilderDelegate(
+                    (context, index) {
+                      DownloadStub stub = items.elementAt(index);
+                      return ExpansionTile(
+                        key: PageStorageKey(stub.id),
+                        leading: AlbumImage(item: stub.baseItem),
+                        title: Text(stub.baseItem?.name ?? stub.name),
+                        trailing: Row(
+                          mainAxisSize: MainAxisSize.min,
+                          children: [
+                            if ((!(stub.baseItemType == BaseItemDtoType.album ||
+                                    stub.baseItemType ==
+                                        BaseItemDtoType.track)) &&
+                                !FinampSettingsHelper.finampSettings.isOffline)
+                              IconButton(
+                                icon: const Icon(Icons.sync),
+                                onPressed: () {
+                                  _downloadsService.resync(stub, null);
+                                },
+                              ),
+                            IconButton(
+                              icon: const Icon(Icons.delete),
+                              onPressed: () =>
+                                  askBeforeDeleteDownloadFromDevice(
+                                      context, stub),
+                            ),
+                          ],
+                        ),
+                        subtitle: ItemFileSize(stub: stub),
+                        tilePadding: const EdgeInsets.symmetric(horizontal: 16),
+                        children: [
+                          if (stub.baseItemType.hasChildren)
+                            DownloadedChildrenList(parent: stub)
```

Comment:
Probably out-of-scope for this PR, but it's a bit strange that we only allow expanding one level deep. Expanding should always work until getting to the actual tracks, so for an artist or genere one should be able to expand the contained albums to see the tracks...

Reply:
Yes, probably out of scope :sweat_smile:
I do already wonder whether they are loaded lazily as expected, or not..